sale_automatic_workflow/models/automatic_workflow_job.py

```diff
@@ -38,7 +38,10 @@ def _validate_sale_orders(self, order_filter):
         _logger.debug('Sale Orders to validate: %s', sales.ids)
         for sale in sales:
             with savepoint(self.env.cr):
+                user_company = self.env.user.company_id
```

Comment:
@grindtildeath IMO you should define a method to use as context manager...
```python
from contextlib import contextmanager

@contexmanager
def force_company(env, company_id)
    user_company = self.env.user.company_id
    self.env.user.company_id = company_id
    try:  
        yield
    finally:
        self.env.user.company_id = user_company
```
Once you have defined the method it can used as follow...
```python
with savepoint(self.env.cr), force_company(sale.company_id):
    self.action_confirm()
```

With this new method the logic to change the current company is factored into a unique method